dxp/demystifying_aui_script_and_liferay-frontend_component_tags.md

```diff
@@ -0,0 +1,114 @@
+# Demystifying `aui:script` and `liferay-frontend:component` tags
+
+On JSP context you may notice a huge number of `aui:script` and `liferay-frontend:component` tags.
+
+## `aui:script` tag
+
+### What it does:
+
+This taglib allow users to use and require another JS module/resources inside a JSP.
+
+This taglib have three possible use cases:
+
+1. Pure JS on the JSP
+2. Import a AUI/YUI module
+3. Import a ES6 module
+
+#### Pure JS on the JSP
+
+By just placing JavaScript code inside an `aui:script` tag, you can take advantage of optimizing scripts and place them to be made available in the final compressed JavaScript bundle, see [combo for more details](https: / /github.com/liferay/liferay-frontend-guidelines/blob/45bad0334cb467e45e68aee322a2a4bc0b6979d3/dxp/javaScript_minification.md#comboservlet). It is worth mentioning that the same rule applies for the following use cases.
+
+#### Import an AUI/YUI module using `use` property
+
+When using the `use` property, the AUI / YUI module listed in the` use` list will be imported to the script data. If you want to know a little more about ScriptData, see this [link on how resources are injected into DXP](https://github.com/liferay/liferay-frontend-guidelines/blob/ea185686db8d562ac11cc2b95d2e9e7d7a0c6547/dxp/resource_injection.md#scriptdata).
+
+You can import several YUI / AUI modules within the same `use` property, as in this example:
+
+```jsp
+<aui:script use="liferay-upload">
+...
+        var liferayUpload = new Liferay.Upload({
+...
+</aui:script>
+```
+
+Note in the AlloyUI components, coming from the alloy-ui repository, classes are made available from the global variable `A`, as in this [code snippet the Menu component is defined](https://github.com/liferay/alloy-ui/blob/master/src/aui-menu/js/aui-menu.js#L39]).
```

Comment:
I'd say we can go pretty light on the AlloyUI documentation, and possibly even omit it. We don't want anybody writing new code that uses them (unlike `aui:script` itself, which we are happy to have people keep on using, for now).

In fact, I think this whole document needs a big warning at the top making it clear which parts of this API are non-deprecated and ok to use, and which parts are legacy and to be avoided.

Reply:
As used on https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/resource_serving.md#javascript-bundle,

I just copied and paste the warning as the example below:


#### Import an AUI/YUI module using the `use` property 

> ⚠️ This is a deprecated use case, though it still works if someone configures it.⚠️


Also, I added a more detailed description on the beginning of the document as the following example:

## IMPORTANT

Infrastructure is sunsetting AlloyUI since when ["The status and direction of the frontend infrastructure in Liferay 7 & DXP"](https://liferay.dev/blogs/-/blogs/the-status-and-direction-of-the-frontend-infrastructure-in-liferay-7-dxp) blog post was released. Also, We are working hard in removing those components on [Remove AUI / YUI from Liferay Portal](https://issues.liferay.com/browse/LPS-98564) Epic. Please, avoid using AlloyUI JS components and tags, except for `aui:script` tag.